app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PostsController < ApplicationController
+  def index
+    @user = User.find(params[:user_id])
+    @users = User.all
+    @post = Post.all
+  end
+
+  def show
+    @user = User.find(params[:user_id])
+    @users = User.all
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+
+    # @post = current_user.posts.build(post_params) # Assuming you have a relationship between users and posts
+    @post = @user.posts.build(post_params) # Assuming you have a relationship between users and posts
+
+    if @post.save
+      # Handle successful creation, e.g., redirect to a show page
+      # redirect_to user_path(@user)
+      redirect_to user_posts_path
```

Comment:
- [ ] Optional: Can you try not to comment too much on the code? I'm just asking you this because of the confusion created. I think it's a good idea to delete them. @benja27